sympy/solvers/tests/test_solvers.py

```diff
@@ -1557,23 +1547,68 @@ def test_issue_14607():
         assert s[0][var].simplify() == knownsolution[var].simplify()
 
 
-@slow
 def test_lambert_multivariate():
     from sympy.abc import a, x, y
-    from sympy.solvers.bivariate import _filtered_gens, _lambert, _solve_lambert
-
+    from sympy.solvers.bivariate import _filtered_gens, _lambert, _power_list
     assert _filtered_gens(Poly(x + 1/x + exp(x) + y), x) == set([x, exp(x)])
     assert _lambert(x, x) == []
     assert solve((x**2 - 2*x + 1).subs(x, log(x) + 3*x)) == [LambertW(3*S.Exp1)/3]
     assert solve((x**2 - 2*x + 1).subs(x, (log(x) + 3*x)**2 - 1)) == \
           [LambertW(3*exp(-sqrt(2)))/3, LambertW(3*exp(sqrt(2)))/3]
     assert solve((x**2 - 2*x - 2).subs(x, log(x) + 3*x)) == \
           [LambertW(3*exp(1 - sqrt(3)))/3, LambertW(3*exp(1 + sqrt(3)))/3]
-    assert solve(x*log(x) + 3*x + 1, x) == [exp(-3 + LambertW(-exp(3)))]
     eq = (x*exp(x) - 3).subs(x, x*exp(x))
     assert solve(eq) == [LambertW(3*exp(-LambertW(3)))]
     # coverage test
     raises(NotImplementedError, lambda: solve(x - sin(x)*log(y - x), x))
+    assert solve(x**3 - 3**x, x) == [3, -3*LambertW(-log(3)/3)/log(3)]
+    assert set(solve(3*log(x) - x*log(3))) == set(  # 2.478... and 3
+        [3, -3*LambertW(-log(3)/3)/log(3)])
+    assert solve(LambertW(2*x) - y, x) == [y*exp(y)/2]
+    # Helper function power_list
+    assert _power_list(x, x) == {x: [1]}
+    assert _power_list(x, y) == {}
+    assert _power_list(x**2 + x*y + 1, x) == {x: [2, 1]}
+    assert _power_list(x**2 + x*y + 1, x, y) == {x: [2, 1], y: [1]}
+    assert _power_list(x**4 + exp(x**5 + x + 1), x) == {x: [1, 4, 5]}
```

Comment:
hmm...this does not look promising at first glance. How will powers inside a function be used and how will the caller know which came from where?

Reply:
@smichr May be this function`_power_list` is not needed anymore. Because all of the motive to create this method has been solved by your suggested code.
```
if lhs.is_Add or lhs.is_Mul:
        lhs = lhs.replace(lambda i:i.is_Pow and i.base == symbol and i.exp%2==0, lambda i: t**i.exp)
```